prompt/prompt-executor/prompt-executor-clients/prompt-executor-google-client/src/commonMain/kotlin/ai/koog/prompt/executor/clients/google/GoogleLLMClient.kt

```diff
@@ -43,10 +43,15 @@ import kotlin.uuid.Uuid
  *
  * @property baseUrl The base URL for the Google AI API.
  * @property timeoutConfig Timeout configuration for API requests.
+ * @property includeThoughts If `true`, requests Gemini to add reasoning blocks to the response.
+ *                          Defaults to `false` to keep responses short and fast.
+ * @property thinkingBudget Hard cap for reasoning tokens.Ignored by models that don't support budgets.
  */
 public class GoogleClientSettings(
     public val baseUrl: String = "https://generativelanguage.googleapis.com",
-    public val timeoutConfig: ConnectionTimeoutConfig = ConnectionTimeoutConfig()
+    public val timeoutConfig: ConnectionTimeoutConfig = ConnectionTimeoutConfig(),
+    public val includeThoughts: Boolean = false,
+    public val thinkingBudget: Int? = null
```

Comment:
It's better to make these as parameters in `Prompt` instead. Current implementation limits the client once it is created only to a certain behavior. But Prompt parameters offer greater flexibility, so that you can reuse existing clients with different generation parameters. Like with temperature for example

Reply:
Moved to LLMParams, thanks :)
Thought about it in the first place, but still feels that Prompt and LLMParams too common place to put google specific logic in.
Feels like LLMParams should be different structure depending on model